### PR TITLE
4. add bundle repo validator

### PR DESCRIPTION
**Pull request by dfrankland**
depends on #109 

Adds validation for git repo details we collect independently from other context we collect

### CODE REVIEW

File: context/src/lib.rs [hunk 1]
```diff
@@ -5,33 +5,4 @@
...
-        FieldLen::TooShort(String::from(trimmed_field))
-    } else if (1..=MAX_LEN).contains(&trimmed_field_len) {
-        FieldLen::Valid
-    } else {
-        FieldLen::TooLong(String::from(safe_truncate_string::<MAX_LEN, _>(
-            &trimmed_field,
-        )))
-    }
-}
+mod string_safety;
```
Comment (dfrankland) [inline context/src/lib.rs:8]: Moved this to its own module for better organization

File: context/src/repo/validator.rs
Comment (dfrankland) [inline context/src/repo/validator.rs:1]: Currently we don't validate this data we collect independently from other pieces of context like env. This helps to check independently whether we have this information before we mash everything together.

File: context/src/string_safety.rs
Comment (dfrankland) [inline context/src/string_safety.rs:1]: moved from `lib.rs` for organization

**Review by riya-n — APPROVED** [review]
nice! lgtm

**Review by gnalh — APPROVED** [review]

**Pull request merged**